### ISSUE TITLE
Fatal 404 when accessing lambda function URL

### ISSUE THREAD

**Issue by Javarome** [netlify/netlify-lambda#478]
**Describe the bug**
When requesting a lambda function, an error stops the server because HTTP headers are written after the response has been sent.

**To Reproduce**
1. Build a lambda function that sets headers in its response
2. Run lambda functions dev server
3. Request the lambda function URL
4. See error:
````
node:internal/process/promises:245
          triggerUncaughtException(err, true /* fromPromise */);
          ^

Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
    at new NodeError (node:internal/errors:329:5)
    at ServerResponse.setHeader (node:_http_outgoing:573:11)
    at callback (/Users/jerome/project/proxy-api/node_modules/netlify-lambda/lib/serve.js:37:16)
    at /Users/jerome/project/proxy-api/node_modules/netlify-lambda/lib/serve.js:72:7
    at processTicksAndRejections (node:internal/process/task_queues:94:5) {
  code: 'ERR_HTTP_HEADERS_SENT'
````

**Expected behavior**
1. The (dev) server should not stop, just display an error.
2. The lambdaResponse headers should be sent only once.

**Desktop (please complete the following information):**
  System:
    OS: macOS 11.5
    CPU: (8) arm64 Apple M1
    Memory: 639.13 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 15.12.0 - /opt/homebrew/bin/node
    Yarn: 1.22.11 - ~/project/proxy-api/node_modules/.bin/yarn
    npm: 7.12.1 - /opt/homebrew/bin/npm
  npmPackages:
    netlify-lambda: ^2.0.12 => 2.0.12

**Comment by Javarome** [comment]
Actually the lambda function is a handler built through `aws-lambda-fastify` wrapping. Debugging more, it looks that the problem is that the `server/` GET request is transmitted to fastify which replies that no `/server` route has been registered (which is true, as other route names are registered, expected to be "behind" the lambda function route), thus responding a early 404. In such a case, it looks that fastify has already replied when `netlify-lambda` express response tries to set headers.

**Comment by Javarome** [comment]
Debugging netlify-lambda's serve.js, it looks that the callback is executed more than once. The first time, the lambdaResponse headers are set on the express response without errors, and the second time of course the "already sent" error appears.

**Comment by erezrokah** [comment]
Hi @Javarome, the `serve` command is deprecated, see https://github.com/netlify/netlify-lambda#netlify-lambda-serve-legacy-command-proxying-for-local-development.

I recommend switching to `netlify dev`, see [here](https://cli.netlify.com/netlify-dev). If you have issues with `netlify dev` you can open [a new issue](https://github.com/netlify/cli/issues/new?assignees=&labels=type%3A+bug&template=bug_report.md&title=Please+replace+with+a+clear+and+descriptive+title) on the Netlify CLI repo.

**Comment by Javarome** [comment]
Okay let's close this issue. Furthermore I am using [aws-lambda-fastify](https://github.com/fastify/aws-lambda-fastify) which might be responsible for the double callback bug.
